api.go

```diff
@@ -637,6 +646,15 @@ func (r *Raft) Leader() ServerAddress {
 // will fail.
 func (r *Raft) Apply(cmd []byte, timeout time.Duration) ApplyFuture {
 	metrics.IncrCounter([]string{"raft", "apply"}, 1)
+
+	return r.ApplyLog(Log{Data: cmd}, timeout)
+}
+
+// ApplyLog performs Apply but takes in a Log directly. The only values
+// currently taken from the submitted Log are Data and Extensions.
+func (r *Raft) ApplyLog(log Log, timeout time.Duration) ApplyFuture {
+	metrics.IncrCounter([]string{"raft", "apply_with_log"}, 1)
```

Comment:
Is there value in making this a new metric name? My concern is that people with existing monitoring setup likely have `raft.apply` graphed and assume it covers all `apply` calls. If their workload slowly transitions over time to include more chunked calls then they won't see those in their existing graphs as the chunks will only get counted as `apply_with_log`.

I wonder if it would be better to just make this `apply` like now and remove the increment from `Apply` above to avoid double counting. We can record a separate counter in Consul if we want to make it clear how many applies are chunked vs not chunked where we actually have context to know.

It seems like an implementation detail of the client whether this should be separate or not so I vote for keeping it just as `apply` and letting clients differentiate in how/what they are applying where they have meaningful context for that. (e.g. Consul could add a `raft.chunked` counter to be explicit about why it's different/important).

Reply:
I'm not sure if there's any value and I'm fine just moving apply there, will do so.